setup_ts.py

```diff
@@ -120,6 +123,24 @@ def download_toolchain():
     shutil.rmtree('toolchain/yarn', ignore_errors=True)
     Path(f'toolchain/yarn-{yarn_version}').rename('toolchain/yarn')
 
+def update_package():
+    if jupyter_lab_version == '2':
+        package_json = json.load(open('ts/jupyter_extension/package.json'))
+        json.dump(package_json, open('ts/jupyter_extension/.package_default.json', 'w'), indent=2)
+
+        package_json['scripts']['build'] = 'tsc && jupyter labextension link .'
+        package_json['dependencies']['@jupyterlab/application'] = '^2.3.0'
+        package_json['dependencies']['@jupyterlab/launcher'] = '^2.3.0'
+
+        package_json['jupyterlab']['outputDir'] = 'build'
+        json.dump(package_json, open('ts/jupyter_extension/package.json', 'w'), indent=2)
+        print(f'updated package.json with {json.dumps(package_json, indent=2)}')
+
+def restore_package():
+    if jupyter_lab_version == '2':
+        package_json = json.load(open('ts/jupyter_extension/.package_default.json'))
```

Comment:
Delete `.package_default.json` or put it into gitignore.

Reply:
done, make sense